tests/functional/end-to-end.bats

```diff
@@ -73,6 +73,22 @@ test_one_check() {
 	test_one_check_expect $1 $2 1
 }
 
+test_parse_error() {
+	local SOURCE_FILENAME=$1
+	local OUTPUT_FILENAME=$2
+	run ${SELINT_PATH} -c configs/default.conf ./policies/parse_errors/${SOURCE_FILENAME}
+	echo ${output}
+	[ "$status" -eq 70 ]
+	local EXPECTED_OUTPUT=$(cat ./policies/parse_errors/${OUTPUT_FILENAME})
+	echo ${EXPECTED_OUTPUT}
+	# compatibility for different bison versions and ignore NOTE output
+	local NORMALIZED_OUTPUT=$(grep -vE '^Note: ' <<< "${output/, expecting \$end/, expecting end of file}")
+	echo ${NORMALIZED_OUTPUT}
+	[ "${NORMALIZED_OUTPUT}" == "${EXPECTED_OUTPUT}" ]
+	run valgrind --leak-check=full --show-leak-kinds=all --errors-for-leak-kinds=all --error-exitcode=23 ${SELINT_PATH} -c configs/default.conf ./policies/parse_errors/${SOURCE_FILENAME}
```

Comment:
As much as I love running valgrind in tests, this is really slowing down the functional tests speed.  Timing with the code as is, vs with this line commented out, we go from 6.4s without it to 31.2s with.

31s is no big deal in a PR, but it's pretty painful if you're doing development and trying to test locally.

I spent a while with bats docs trying to figure out the best way to selectively run this portion of the test only on PRs or when explicitly requested, and I couldn't come up with anything super clean.  I have a change locally that moves this test (and the broken config one which similarly runs valgrind on an error condition) into a separate bats file, so PRs and interested parties can run the set of bats tests they want.  It may be time to split into separate files anyways, since 47 tests (and growing) in a single file is a lot.  The main problem there is that I'd prefer to run at least non-valgrind versions of these regularly, and things start getting messy to support that.

Ideally, I'd like to be able to just pass an argument to bats that sets a global variable in our tests and we can check it and only selectively test valgrind, but if bats has that functionality, I can't find it.

Is there a better solution that I'm missing?